src/java/org/apache/cassandra/cql3/statements/schema/CommentOnTableStatement.java

```diff
@@ -0,0 +1,140 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.cql3.statements.schema;
+
+import org.apache.cassandra.audit.AuditLogContext;
+import org.apache.cassandra.audit.AuditLogEntryType;
+import org.apache.cassandra.auth.Permission;
+import org.apache.cassandra.cql3.CQLStatement;
+import org.apache.cassandra.cql3.QualifiedName;
+import org.apache.cassandra.cql3.statements.SchemaDescriptionsUtil;
+import org.apache.cassandra.schema.KeyspaceMetadata;
+import org.apache.cassandra.schema.Keyspaces;
+import org.apache.cassandra.schema.Keyspaces.KeyspacesDiff;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.schema.TableParams;
+import org.apache.cassandra.service.ClientState;
+import org.apache.cassandra.tcm.ClusterMetadata;
+import org.apache.cassandra.transport.Event.SchemaChange;
+import org.apache.cassandra.transport.Event.SchemaChange.Change;
+import org.apache.cassandra.transport.Event.SchemaChange.Target;
+
+/**
+ * Handles the execution of COMMENT ON TABLE CQL statements.
+ * <p>
+ * This statement allows users to add descriptive comments to tables for documentation purposes.
+ * Comments are stored in the table metadata and displayed when using DESCRIBE TABLE.
+ * </p>
+ * <p>
+ * Syntax: {@code COMMENT ON TABLE [keyspace_name.]table_name IS 'comment text';}
+ * </p>
+ * <p>
+ * Comments can be removed by setting them to an empty string or null.
+ * </p>
+ *
+ * @see CommentOnKeyspaceStatement
+ * @see CommentOnColumnStatement
+ * @see CommentOnTypeStatement
+ */
+public final class CommentOnTableStatement extends AlterSchemaStatement
+{
+    private final String tableName;
+    private final String comment;
+
+    public CommentOnTableStatement(String keyspaceName, String tableName, String comment)
+    {
+        super(keyspaceName);
+        this.tableName = tableName;
+        this.comment = comment;
+    }
+
+    @Override
+    public void validate(ClientState state)
+    {
+        super.validate(state);
+        SchemaDescriptionsUtil.validateComment(comment);
+    }
+
+    @Override
+    public Keyspaces apply(ClusterMetadata metadata)
+    {
+        Keyspaces schema = metadata.schema.getKeyspaces();
+        KeyspaceMetadata keyspace = schema.getNullable(keyspaceName);
+
+        if (null == keyspace)
+            throw ire("Keyspace '%s' doesn't exist", keyspaceName);
+
+        TableMetadata table = keyspace.getTableOrViewNullable(tableName);
+        if (null == table)
+            throw ire("Table '%s.%s' doesn't exist", keyspaceName, tableName);
+
+        if (table.isView())
+            throw ire("Cannot set comment on materialized view '%s.%s'. Comments should be set on the base table.", keyspaceName, tableName);
+
```

Comment:
To @smiklosovic 's point, I think you only want to allow comment on regular table, i.e. reject if `table.kind != TableMetadata.Kind.REGULAR`